crates/aptos-rosetta/src/block.rs

```diff
@@ -68,64 +60,69 @@ async fn block(request: BlockRequest, server_context: RosettaContext) -> ApiResu
 /// Build up the transaction, which should contain the `operations` as the change set
 async fn build_block(
     parent_block_identifier: BlockIdentifier,
-    block_info: BlockInfo,
-    transactions: Vec<aptos_rest_client::Transaction>,
+    block: aptos_rest_client::aptos_api_types::Block,
 ) -> ApiResult<Block> {
     // note: timestamps are in microseconds, so we convert to milliseconds
-    let timestamp = get_timestamp(block_info);
-    let block_identifier = BlockIdentifier::from_block_info(block_info);
+    let timestamp = get_timestamp(block.block_timestamp.0);
+    let block_identifier = BlockIdentifier::from_block(&block);
 
     // Convert the transactions and build the block
-    let mut txns: Vec<Transaction> = Vec::new();
-    for txn in transactions {
-        txns.push(Transaction::from_transaction(txn).await?)
+    let mut transactions: Vec<Transaction> = Vec::new();
+    if let Some(txns) = block.transactions {
+        for txn in txns {
+            transactions.push(Transaction::from_transaction(txn).await?)
+        }
     }
 
     Ok(Block {
         block_identifier,
         parent_block_identifier,
         timestamp,
-        transactions: txns,
+        transactions,
     })
 }
 
 /// Retrieves a block by its index
 async fn get_block_by_index(
```

Comment:
"by_height" if possible :D

Reply:
Yeah, I will fix it all when my other PR gets through